routing/cross_mwm_index_graph.cpp

```diff
@@ -156,55 +160,66 @@ void CrossMwmIndexGraph::GetTwins(Segment const & s, bool isOutgoing, vector<Seg
 {
   CHECK(IsTransition(s, isOutgoing), ("The segment", s, "is not a transition segment for isOutgoing ==", isOutgoing));
   // Note. There's an extremely rare case when a segment is ingoing and outgoing at the same time.
-  // |twins| is not filled for such cases. For details please see a note in
-  // rossMwmIndexGraph::GetEdgeList().
+  // |twins| is not filled for such cases. For details please see a note in CrossMwmIndexGraph::GetEdgeList().
   if (IsTransition(s, !isOutgoing))
     return;
 
   twins.clear();
-  // @TODO(bykoianko) It's necessary to check if mwm of |s| contains an A* cross mwm section
-  // and if so to use it. If not, osrm cross mwm sections should be used.
-
-  auto const getTwins = [&](TRoutingMappingPtr const & segMapping) {
-    vector<string> const & neighboringMwm = segMapping->m_crossContext.GetNeighboringMwmList();
 
-    for (string const & name : neighboringMwm)
-      InsertWholeMwmTransitionSegments(m_numMwmIds->GetId(CountryFile(name)));
-
-    auto it = m_transitionCache.find(s.GetMwmId());
-    CHECK(it != m_transitionCache.cend(), ());
-
-    vector<ms::LatLon> const & latLons =
-        isOutgoing ? GetLatLon(it->second.m_outgoing, s) : GetLatLon(it->second.m_ingoing, s);
-    for (string const & name : neighboringMwm)
-    {
-      NumMwmId const numMwmId = m_numMwmIds->GetId(CountryFile(name));
-      auto const addFirstValidSegment = [&](TRoutingMappingPtr const & mapping) {
-        for (auto const & ll : latLons)
+  // Note. The code below looks for twins based on geometry index. This code works for
+  // any combination mwms with different cross mwm sections.
+  // It's possible to implement a faster version for two special cases:
+  // * all neighbouring mwms have cross_mwm section
```

Comment:
Done, however there're two variants of writing the word "neighboring".
http://grammarist.com/spelling/neighbor-neighbour/

Neighbor is preferred in American English, and neighbour is preferred in all the other main varieties of English. The words are the same in every other respect. The spelling difference extends to derivative words such as neighborhood/neighbourhood, neighborly/neighbourly, neighbored/neighboured, and neighboring/neighbouring.